### PR TITLE
WIP: Add mysql serverspec

### PR DESCRIPTION
**Pull request by pwplusnick**
This adds the mysql serverspec and sets the acl
for the mysql log files, which must have been
overlooked earlier.

### CODE REVIEW

File: roles/percona-server/templates/etc/serverspec/mysql_spec.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+         '  missingok',
+         '  rotate 7',
+         '  compress',
+         '  postrotate',
+         '  test -x /usr/bin/mysqladmin && /usr/bin/mysqladmin ping > /dev/null && /usr/bin/mysqladmin flush-logs > /dev/null',
+         '  endscript',
+         '  minsize 100k',
+         '}']
+  file_contents.each do |file_line|
+    it { should contain file_line}
```
Comment (pwplusnick) [inline roles/percona-server/templates/etc/serverspec/mysql_spec.rb]: whitespace

File: roles/percona-server/templates/etc/serverspec/mysql_spec.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+         '{',
+         '  daily',
+         '  missingok',
+         '  rotate 7',
+         '  compress',
+         '  postrotate',
+         '  test -x /usr/bin/mysqladmin && /usr/bin/mysqladmin ping > /dev/null && /usr/bin/mysqladmin flush-logs > /dev/null',
+         '  endscript',
+         '  minsize 100k',
+         '}']
```
Comment (pwplusnick) [inline roles/percona-server/templates/etc/serverspec/mysql_spec.rb]: clean up this eyesore

File: roles/percona-server/tasks/main.yml [hunk 3]
```diff
@@ -126,3 +136,7 @@
     - logrotate
     - logging
   when: logging.enabled|default('True')|bool
+
+- include: serverspec.yml
+  tags:
+    - serverspec
```
Comment (pwplusnick) [inline roles/percona-server/tasks/main.yml]: newline

**Comment by pwplusnick** [comment]
Those comments may seem terse, but I was just making notes for myself :)

File: roles/percona-server/tasks/serverspec.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+- name: serverspec checks for keystone
```
Comment (vlope093) [inline roles/percona-server/tasks/serverspec.yml]: Change it to `serverspec checks for mysql`

Comment (pwplusnick) [inline roles/percona-server/tasks/serverspec.yml]: good catch!

File: roles/percona-server/tasks/main.yml (hunk 3 above)
Comment (j2sol) [inline roles/percona-server/tasks/main.yml]: This needs a conditional.

File: roles/percona-server/tasks/serverspec.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+- name: serverspec checks for mysql
+  template: src={{ item }}
+            dest=/etc/serverspec/spec/localhost/
+            mode=0755
+  with_fileglob: ../templates/etc/serverspec/*
```
Comment (j2sol) [inline roles/percona-server/tasks/serverspec.yml]: newline

File: roles/percona-server/templates/etc/serverspec/mysql_spec.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+files = ['bind-inaddr-any.cnf', 'replication.cnf', 'tuning.cnf' , 'utf8.cnf']
+files.each do |file|
+  describe file("/etc/mysql/conf.d/#{file}") do
+    it { should be_mode 644 }
+    it { should be_owned_by 'root' }
+    it { should be_grouped_into 'root' }
+  end
+end 
+
+files = {'conf.d' => 755, 'debian.cnf'=> 600, 'my.cnf'=>644 }
```
Comment (j2sol) [inline roles/percona-server/templates/etc/serverspec/mysql_spec.rb]: whitespace.

**Comment by j2sol** [comment]
ok to test

File: roles/percona-server/templates/etc/serverspec/mysql_spec.rb [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+files.each do |file, mode|
+  describe file("/etc/mysql/#{file}") do
+    it { should be_mode mode }
+    it { should be_owned_by 'root' }
+    it { should be_grouped_into 'root' }
+  end
+end
+
+files = ['mysql.err', 'mysql.log' ]
+files.each do |file, mode|
```
Comment (vlope093) [inline roles/percona-server/templates/etc/serverspec/mysql_spec.rb]: @pwplusnick you don't need the mode in this line

**Pull request merged**